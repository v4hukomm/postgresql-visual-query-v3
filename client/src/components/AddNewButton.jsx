import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import React from 'react';
import * as PropTypes from 'prop-types';
import { Button } from 'reactstrap';

export function AddNewButton({ id, size, onClick }) {
  return (
    <Button
      className=""
      outline
      color="info"
      size={size}
      id={id}
      onClick={onClick}
    >
      <FontAwesomeIcon icon="plus" />
    </Button>
  );
}

AddNewButton.propTypes = {
  id: PropTypes.string,
  size: PropTypes.string,
  onClick: PropTypes.func,
};
