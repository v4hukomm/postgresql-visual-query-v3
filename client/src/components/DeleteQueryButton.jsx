import React from 'react';
import { Button } from 'reactstrap';
import { connect } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import * as PropTypes from 'prop-types';
import { deleteQuery, setActiveQuery } from '../actions/queryActions';
import { deleteQueries, removeMainFromQueries } from '../actions/queriesActions';

export const DeleteQueryButton = (props) => {
  const handleOnClick = () => {
    props.deleteQuery();

    const mainQuery = props.queries.find(query => query.id === 0);

    if (mainQuery) {
      props.removeMainFromQueries();
      props.setActiveQuery(mainQuery);
    } else {
      props.deleteQueries();
    }
  };

  return (
    <Button color="danger" className="btn-sm" onClick={handleOnClick} style={{ borderTopLeftRadius: '0px', borderBottomLeftRadius: '0px' }}>
      <FontAwesomeIcon icon="times" />
    </Button>
  );
};

DeleteQueryButton.propTypes = {
  deleteQuery: PropTypes.func,
  deleteQueries: PropTypes.func,
  queries: PropTypes.arrayOf(PropTypes.shape({})),
  removeMainFromQueries: PropTypes.func,
  setActiveQuery: PropTypes.func,
};

const mapStateToProps = store => ({
  queries: store.queries,
});

const mapDispatchToProps = {
  deleteQuery,
  deleteQueries,
  removeMainFromQueries,
  setActiveQuery,
};

export default connect(mapStateToProps, mapDispatchToProps)(DeleteQueryButton);
