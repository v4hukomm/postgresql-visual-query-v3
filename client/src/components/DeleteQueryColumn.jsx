import React, { useState } from 'react';
import { connect } from 'react-redux';
import { 
    Button,
    CustomInput,
    Col,
    Input,
    InputGroup,
    Row, 
} from 'reactstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { updateColumn, removeColumn } from '../actions/queryActions';

export const DeleteQueryColumn = (props) => {

    const [returning, setReturning] = useState(props.data.returning);
    const [returningOnly, setReturningOnly] = useState(props.data.returning);

    let column = props.data;

    const handleChange = (e) => {
        column.column_filters.map((filter) => {
            if (filter.id === parseInt(e.target.name)) {
                filter.filter = e.target.value;
            }
            return filter;
            });
    };

    const handleSave = () => {
        props.updateColumn(column);
    };

    const handleRemoveColumn = () => {
        props.removeColumn(props.data);
    };

    const handleReturningSwitch = () => {
        column.returning = !column.returning;
        setReturning(current => !current);

        props.updateColumn(column);
    };  

    const handleReturningOnlySwitch = () => {
        column.returningOnly = !column.returningOnly;
        setReturningOnly(current => !current);

        props.updateColumn(column);
    };
    
    return (
        <div>
            <div className='border-bottom border-right pr-2 pl-1 border-dark'>
                        <h6 className='d-inline'>
                            {column.table_name}.{column.column_name}
                        </h6>
                        <Button
                        className="mr-2 float-right"
                        size="sm"
                        color="danger"
                        onClick={handleRemoveColumn}
                        >
                        <FontAwesomeIcon icon="times" />
                        </Button>
                        <InputGroup size="sm">
                        <CustomInput
                         className=""
                         type="switch"
                         id={`column-returning-${column.id}`}
                         key={column.id}
                         name="returning"
                         disabled={props.returning || returningOnly}
                         checked={returning}
                         onChange={handleReturningSwitch}
                         label="Returning">
                        </CustomInput>
                        <CustomInput
                         className=""
                         type="switch"
                         id={`column-returning-only-${column.id}`}
                         key={column.id}
                         name="returning-only"
                         disabled={props.returning}
                         checked={returningOnly}
                         onChange={handleReturningOnlySwitch}
                         label="Returning only">
                        </CustomInput>
                        </InputGroup>
                </div>
                        <Col className='mr-2 border-bottom'>
                            {column.column_filters.map((filter) => (
                            <Row className="pt-2 pb-2 pr-2 pl-1 border-top" key={filter.id}>
                                <Input
                                 id={`column-filter-${filter.id}`}
                                 name={filter.id}
                                 key={filter.id}
                                 type="text"
                                 placeholder="Ex. = 3"
                                 defaultValue={filter.filter}
                                 onBlur={handleSave}
                                 onChange={handleChange}
                                 disabled={returningOnly}>
                               </Input>
                               <p className='ml-2'>AND</p>
                            </Row>
                            ))}
                        </Col>
        </div>
    );
};

const mapStateToProps = (store) => ({
    returning: store.query.returning,
});

const mapDispatchToProps = {
    updateColumn,
    removeColumn,
};


export default connect(mapStateToProps, mapDispatchToProps)(DeleteQueryColumn);