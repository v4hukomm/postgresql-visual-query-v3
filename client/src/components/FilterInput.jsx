import React from 'react';
import { connect } from 'react-redux';
import {
  Input,
} from 'reactstrap';
import { updateColumnFilter } from '../actions/queryActions';

export const FilterInput = (props) => {
  let filterValue = props.value;

  const handleChange = (e) => {
    filterValue = e.target.value;
  };

  const handleSave = () => {
    const data = {
      filterId: props.filterId,
      columnId: props.columnId,
      filter: filterValue,
    };
    props.updateColumnFilter(data);
  };

  return (
    <Input
      id={`column-filter-${props.filterId}`}
      name={props.filterId}
      key={props.filterId}
      type="text"
      placeholder="Ex. = 3"
      defaultValue={props.value}
      onBlur={handleSave}
      onChange={handleChange}
      disabled={props.returningOnly}
    />
  );
};

const mapDispatchToProps = {
  updateColumnFilter,
};

export default connect(null, mapDispatchToProps)(FilterInput);
