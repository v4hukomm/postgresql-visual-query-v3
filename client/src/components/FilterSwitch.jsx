import React, { useState } from 'react';
import { connect } from 'react-redux';
import {
    CustomInput,
} from 'reactstrap';
import { updateColumn } from '../actions/queryActions';

export const FilterSwitch = (props) => {

    const [returning, setReturning] = useState(props.column.returning);
    const [returningOnly, setReturningOnly] = useState(props.column.returningOnly);

    let column = props.column;

    const handleChange = () => {
        if (props.only) {
            column.returningOnly = !column.returningOnly;
            setReturningOnly(current => !current);
            props.updateColumn(column);

        } else {
            column.returning = !column.returning;
            setReturning(current => !current);
            props.updateColumn(column);
        }
    };
    
    return (
        <CustomInput
            id={`column-${props.only ? 'returningOnly' : 'returning'}-${column.id}`}
            name={(props.only ? 'returningOnly' : 'returning')}
            key={column.id}
            type="switch"
            defaultValue={(props.only ? returningOnly : returning)}
            checked={(props.only ? returningOnly : returning)}
            disabled={props.only ? false : props.returningAll}
            onChange={handleChange}
        />
    );
};

const mapDispatchToProps = {
    updateColumn,
};

const mapStateToProps = (store) => ({
    returningAll: store.query.returning,
});

export default connect(mapStateToProps, mapDispatchToProps)(FilterSwitch);