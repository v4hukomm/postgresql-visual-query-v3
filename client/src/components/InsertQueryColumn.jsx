import React, { useState } from 'react';
import { connect } from 'react-redux';
import { 
    Button,
    CustomInput,
    Col,
    Input,
    InputGroup,
    Row, 
} from 'reactstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { updateColumn, removeColumn } from '../actions/queryActions';

export const InsertQueryColumn = (props) => {

    const [returning, setReturning] = useState(props.data.returning);

    let column = props.data;
    let rowCount = Array.from({ length: props.rows }, (_, index) => index);

    const handleChange = (e) => {
        column.column_values.map((value) => {
            if (value.id === parseInt(e.target.name)) {
                value.value = e.target.value;
            }
            return value;
            });
    };

    const handleSave = () => {
        props.updateColumn(column);
    };

    const handleRemoveColumn = () => {
        props.removeColumn(props.data);
    };

    const handleSwitch = () => {
        column.returning = !column.returning;
        setReturning(current => !current);

        props.updateColumn(column);
    };

    return (
        <div>
            <h6 className='ml-4 d-inline'>{column.column_name}</h6>
            <Button
                className="ml-3"
                size="sm"
                color="danger"
                onClick={handleRemoveColumn}
                >
                <FontAwesomeIcon icon="times" />
            </Button>
            <InputGroup size="sm">
            <CustomInput
                className="mr-4 ml-4"
                type="switch"
                id={`column-returning-${column.id}`}
                key={column.id}
                name="returning"
                checked={returning}
                onChange={handleSwitch}
                label="Returning">
            </CustomInput>
            </InputGroup>
            <Col sm='1' className='mr-4 ml-4'>
                {rowCount.map(i =>
                <Row key={`column-values-row-${i}`}>
                    <Input className='mt-2 mb-2'
                    key={`column-value-${i}`}
                    name={i}
                    type="text"
                    placeholder="filter"
                    defaultValue={props.data.column_values[i].value}
                    onBlur={handleSave}
                    onChange={handleChange}
                    disabled={props.fromQuery}>
                    </Input>
                </Row>
                    )}
            </Col>
        </div>
    );
};

const mapStateToProps = (store) => ({
    fromQuery: store.query.fromQuery,
    rows: store.query.rows,
});

const mapDispatchToProps = {
    updateColumn,
    removeColumn,
};


export default connect(mapStateToProps, mapDispatchToProps)(InsertQueryColumn);