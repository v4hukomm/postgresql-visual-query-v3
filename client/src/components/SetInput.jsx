import React from 'react';
import { connect } from 'react-redux';
import {
  Input,
} from 'reactstrap';
import { updateColumn } from '../actions/queryActions';

export const SetInput = (props) => {
  const { column } = props;

  const handleChange = (e) => {
    column.column_value = e.target.value;
  };

  const handleSave = () => {
    props.updateColumn(column);
  };

  return (
    <Input
      id={`column-set-${props.column.id}`}
      name="column_set"
      key={props.column.id}
      type="text"
      placeholder="NULL"
      defaultValue=""
      onBlur={handleSave}
      onChange={handleChange}
      disabled={!props.enabled}
    />
  );
};

const mapDispatchToProps = {
  updateColumn,
};

export default connect(null, mapDispatchToProps)(SetInput);
