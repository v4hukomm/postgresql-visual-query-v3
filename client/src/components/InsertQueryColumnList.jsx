import React, { useState } from 'react';
import { connect } from 'react-redux';
import { Button, ButtonDropdown, DropdownToggle, DropdownItem, DropdownMenu, CustomInput, Label, Table } from 'reactstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import RemoveColumnButton from './RemoveColumnButton';
import FilterSwitch from './FilterSwitch';
import ValueInput from './ValueInput';
import { addRows, removeRows, switchFromQuery, updateFromQuery, switchReturning, changeDefaultValue } from '../actions/queryActions';
import { getCorrectQueryName } from '../utils/getCorrectQueryName';

export const InsertQueryColumnList = ({
  columns, addRows, removeRows, queries, switchFromQuery,
  fromQuery, updateFromQuery, switchReturning, returning,
  language, defaultValue, changeDefaultValue,
}) => {
  const [dropDownOpen, setDropDownOpen] = useState(false);

  const handleAddRow = () => {
    addRows();
  };

  const handleRemoveRow = () => {
    removeRows();
  };

  const handleDropDown = () => {
    setDropDownOpen(current => !current);
  };

  const handleSave = (e) => {
    let subQuery;

    if (e.currentTarget.name === 'subqueryDefault') {
      subQuery = {
        subqueryId: 0,
        subquerySql: '',
      };
    } else if (e.currentTarget.name === 'subqueryId') {
      const subqueryId = +e.target.value;
      const subquerySql = queries.find(query => query.id === subqueryId).sql;
      subQuery = {
        subqueryId,
        subquerySql,
      };
    }

    updateFromQuery(subQuery);
  };

  return (
    <div className="mt-2">
      <Label>Add row</Label>
      <Button
        className="mb-1 ml-2"
        outline
        color="info"
        size="sm"
        disabled={fromQuery}
        onClick={handleAddRow}
      >
        <FontAwesomeIcon icon="plus" />
      </Button>
      <Label className="ml-2">Remove last row</Label>
      <Button
        className="mb-1 ml-2"
        outline
        color="info"
        size="sm"
        disabled={fromQuery}
        onClick={handleRemoveRow}
      >
        <FontAwesomeIcon icon="times" />
      </Button>
      <CustomInput
        inline
        className="mr-2 ml-2"
        type="switch"
        id="defaultValue"
        checked={defaultValue !== 'DEFAULT'}
        onChange={changeDefaultValue}
        label="Change DEFAULT to NULL"
      />
      <CustomInput
        inline
        className="mr-2 ml-2"
        type="switch"
        id="returning"
        checked={returning}
        onChange={switchReturning}
        label="Returning all"
      />
      <CustomInput
        inline
        className="mr-2 ml-2"
        type="switch"
        id="fromQuery"
        label="Insert from query"
        checked={fromQuery}
        onChange={switchFromQuery}
      />
      {fromQuery && (
      <ButtonDropdown
        isOpen={dropDownOpen}
        toggle={handleDropDown}
      >
        <DropdownToggle
          className="btn-sm btn-light btn-outline-secondary"
          style={{ borderColor: '#d3d8de' }}
          caret
        >
          QueryName
        </DropdownToggle>
        <DropdownMenu>
          <DropdownItem
            key="query-link-SQ"
            name="subqueryDefault"
            value=""
            onClick={handleSave}
          >
            LinkQuery
          </DropdownItem>
          {queries.map((query, index) => (
            <DropdownItem
              key={`query-${query.id}-column`}
              id={`subquerySql-${index}`}
              name="subqueryId"
              value={query.id}
              onClick={handleSave}
            >
              {getCorrectQueryName(language, query.queryName, query.id)}
            </DropdownItem>
          ))}
        </DropdownMenu>
      </ButtonDropdown>
      )}
      {!!columns.length
            && (
            <Table style={{ width: 'auto' }}>
              <thead>
                <tr>
                  <th className="border-right">Column name</th>
                  {columns.map((column) => (
                    <th className="border-right">
                      {column.table_name}
                      .
                      {column.column_name}
                      <RemoveColumnButton
                        column={column}
                      />
                    </th>
                  ))}
                </tr>
                <tr>
                  <th className="border-right">Returning</th>
                  {columns.map((column) => (
                    <td className="border-right">
                      <FilterSwitch
                        column={column}
                        only={false}
                      />
                    </td>
                  ))}
                </tr>
                <tr>
                  <th className="border-right">Returning only</th>
                  {columns.map((column) => (
                    <td className="border-right">
                      <FilterSwitch
                        column={column}
                        only
                      />
                    </td>
                  ))}
                </tr>
              </thead>
              <tbody>
                {!fromQuery && columns[0].column_values.map((_, index) => (
                  <tr>
                    <td className="text-right">{index === 0 ? 'VALUES' : ''}</td>
                    {columns.map((column) => (
                      <td>
                        <ValueInput
                          column={column}
                          index={index}
                          fromQuery={fromQuery}
                          returningOnly={column.returningOnly}
                        />
                      </td>
                    ))}
                  </tr>
                ))}
              </tbody>
            </Table>
            )}
    </div>
  );
};

const mapStateToProps = (store) => ({
  columns: store.query.columns,
  rows: store.query.rows,
  queries: store.queries,
  fromQuery: store.query.fromQuery,
  returning: store.query.returning,
  language: store.settings.language,
  defaultValue: store.query.defaultValue,
});

const mapDispatchToProps = {
  addRows,
  removeRows,
  switchFromQuery,
  updateFromQuery,
  switchReturning,
  changeDefaultValue,
};

export default connect(mapStateToProps, mapDispatchToProps)(InsertQueryColumnList);
