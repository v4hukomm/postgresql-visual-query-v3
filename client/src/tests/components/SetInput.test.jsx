import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import thunk from 'redux-thunk';
import { SetInput } from '../../components/SetInput';

describe('Component: SetInput', () => {
    let props;
    let mockStore;
    let component;
  
    beforeEach(() => {
    props = {
        column: {
            id: 0,
            column_value: 'test'
        },
        enabled: true,
        updateColumn: jest.fn(),
    }
  
      mockStore = configureStore([thunk]);
    });

    test('SetInput renders with default props', () => {
        component = shallow(<SetInput {...props} />);
    
        expect(component).toMatchSnapshot();
    });

    test('SetInput handleSave calls updateColumn once', () => {
        component = shallow(<SetInput {...props} />).simulate('blur');

        expect(props.updateColumn.mock.calls.length).toBe(1);
    });
});