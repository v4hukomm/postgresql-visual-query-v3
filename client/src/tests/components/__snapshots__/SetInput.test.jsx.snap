// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: SetInput SetInput renders with default props 1`] = `ShallowWrapper {}`;
